src/acvp.c

```diff
@@ -1196,6 +1195,402 @@ static ACVP_RESULT acvp_add_drbg_prereq_val(ACVP_DRBG_CAP_MODE *drbg_cap_mode,
 }
 
 /*
+ * Add top level RSA keygen parameters
+ */
+static ACVP_RESULT acvp_add_rsa_keygen_parm (
+                             ACVP_RSA_CAP_MODE_LIST  *rsa_cap_mode_list,
+                             ACVP_RSA_PARM       param,
+                             int                  value
+                             )
+{
+    switch (param) {
+    case ACVP_PUB_EXP:
+        rsa_cap_mode_list->cap_mode_attrs.keygen->pub_exp = value;
+        break;
+    case ACVP_RAND_PQ:
+        rsa_cap_mode_list->cap_mode_attrs.keygen->rand_pq = value;
+        break;
+    case ACVP_RSA_INFO_GEN_BY_SERVER:
+        rsa_cap_mode_list->cap_mode_attrs.keygen->info_gen_by_server = value;
+        break;
+    default:
+        return ACVP_INVALID_ARG;
```

Comment:
I know we aren't real consistent on this, but an ACVP_LOG_ERR may help debug here.